cfgmgr/sflowmgr.cpp

```diff
@@ -128,18 +130,27 @@ void SflowMgr::sflowHandleSessionAll(bool enable)
 {
     for (auto it: m_sflowPortConfMap)
     {
-        if (!it.second.local_conf)
```

Comment:
In configuring `admin_state` field of CONFIG_DB `SFLOW|global` table, both `sflowHandleSessionAll` and `sflowHandleSessionLocal` will be called if `m_intfAllConf` is true (default value). Before your change, `it.second.local_conf`, to my understanding, controls the mutual exclusion that an individual port is set in either `sflowHandleSessionAll` or `sflowHandleSessionLocal`, depending on whether it has its own configuration or not. Now you remove this variable, a port having its own config can be set twice to APPL_DB by both `sflowHandleSessionAll` and `sflowHandleSessionLocal`, calling `m_appSflowSessionTable.set(it.first, fvs)`.

Reply:
@wangxin - Are you asking a question or suggesting a change?

I replaced `port_info.local_conf` with two variables `port_info.local_rate_cfg` and `port_info.local_admin_cfg` to differeniate between a local change to the port's admin-state and a local change to the port's sampling-rate. This change is needed to address the issue at hand.  Without it we will hit a case where the user configured a port's sampling-rate, but the port will not obey the global admin-status (because there is a local config), as the unfixed UT demonstrates.

> Now you remove this variable, a port having its own config can be set twice to APPL_DB by both sflowHandleSessionAll and sflowHandleSessionLocal, calling m_appSflowSessionTable.set(it.first, fvs).

Not sure what you mean here. If a port has a local config (either sample-rate, admin-status, or both), APPL_DB will be set when either `sflowHandleSessionAll` or `sFlowHandleSessionLocal` is called. This behavior is exactly the same as when there was only one variable `port_info.local_config`, because we are doing a OR (||) check:

https://github.com/Azure/sonic-swss/pull/1224/files#diff-e3e7c0a4d48317ec70d5339d6fd3c6d1R136
https://github.com/Azure/sonic-swss/pull/1224/files#diff-e3e7c0a4d48317ec70d5339d6fd3c6d1R162

Reply:
As you said in reply "APPL_DB will be set when either sflowHandleSessionAll or sFlowHandleSessionLocal is called", APPL_DB will be set twice here https://github.com/Azure/sonic-swss/blob/master/cfgmgr/sflowmgr.cpp#L289

Reply:
Yes, once per function call and only if `m_intfallConf` evaluates to `true`. This is needed. I'm still not sure if you're trying to make a suggestion or asking a question...